tests/integration/steps/resource.go

```diff
@@ -98,6 +98,25 @@ func NamespaceIsPresent(ctx context.Context, name, shouldBePresent string) error
 	}, testcontext.GetRetryOpts()...)
 }
 
+func NamespaceHasLabelAndAnnotation(ctx context.Context, name, label, annotation string) error {
+	k8sClient, err := testcontext.GetK8sClientFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	var ns corev1.Namespace
+	return retry.Do(func() error {
+		err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: name}, &ns)
```

Comment:
I think we should check err, because I think we can run into a panic by calling ns.Labels[label] when the ns is not found.